packages/node-experimental/src/sdk/api.ts

```diff
@@ -0,0 +1,225 @@
+
+  if (maxBreadcrumbs && maxBreadcrumbs <= 0) return;
+
+  const timestamp = dateTimestampInSeconds();
+  const mergedBreadcrumb = { timestamp, ...breadcrumb };
+  const finalBreadcrumb = beforeBreadcrumb
+    ? (consoleSandbox(() => beforeBreadcrumb(mergedBreadcrumb, hint)) as Breadcrumb | null)
+    : mergedBreadcrumb;
+
+  if (finalBreadcrumb === null) return;
+
+  if (client.emit) {
+    client.emit('beforeAddBreadcrumb', finalBreadcrumb, hint);
+  }
+
+  getIsolationScope().addBreadcrumb(finalBreadcrumb, maxBreadcrumbs);
+}
+
+/**
+ * Add a global event processor.
+ */
+export function addGlobalEventProcessor(eventProcessor: EventProcessor): void {
+  getGlobalScope().addEventProcessor(eventProcessor);
+}
+
+/**
+ * Add an event processor to the current isolation scope..
+ */
+export function addEventProcessor(eventProcessor: EventProcessor): void {
+  getIsolationScope().addEventProcessor(eventProcessor);
+}
+
+/** Set the user for the current isolation scope. */
+export function setUser(user: User | null): void {
+  getIsolationScope().setUser(user);
+}
+
+/** Set tags for the current isolation scope. */
+export function setTags(tags: { [key: string]: Primitive }): void {
+  getIsolationScope().setTags(tags);
+}
+
+/** Set a single tag user for the current isolation scope. */
+export function setTag(key: string, value: Primitive): void {
+  getIsolationScope().setTag(key, value);
+}
+
+/** Set extra data for the current isolation scope. */
+export function setExtra(key: string, extra: Extra): void {
+  getIsolationScope().setExtra(key, extra);
+}
+
+/** Set multiple extra data for the current isolation scope. */
+export function setExtras(extras: Extras): void {
+  getIsolationScope().setExtras(extras);
+}
+
+/** Set context data for the current isolation scope. */
+export function setContext(
+  name: string,
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  context: { [key: string]: any } | null,
+): void {
+  getIsolationScope().setContext(name, context);
+}
+
+/** Start a session on the current isolation scope. */
+export function startSession(context?: Session): Session {
+  const client = getClient();
+  const scope = getIsolationScope();
+
+  const { release, environment = DEFAULT_ENVIRONMENT } = client.getOptions();
+
+  // Will fetch userAgent if called from browser sdk
+  const { userAgent } = GLOBAL_OBJ.navigator || {};
+
+  const session = makeSession({
+    release,
+    environment,
+    user: scope.getUser(),
+    ...(userAgent && { userAgent }),
+    ...context,
+  });
+
+  // End existing session if there's one
+  const currentSession = scope.getSession && scope.getSession();
+  if (currentSession && currentSession.status === 'ok') {
+    updateSession(currentSession, { status: 'exited' });
+  }
+  endSession();
+
+  // Afterwards we set the new session on the scope
+  scope.setSession(session);
+
+  return session;
+}
+
+/** End the session on the current isolation scope. */
```

Comment:
I feel like this should end the session on the current scope and not isolation scope. Or at least look upwards in the scope stack.

Reply:
that's a valid point, but we don't really know the scope stack right now - only what the current active scope/isolation scope is 😬 but I think we can/should overall tweak this a bit more in follow ups...

Reply:
FWIW this currently lives on the hub, so if that is forked I guess that will also kind of not work anymore?

Reply:
Interesting. I guess it's fine then but it's behaviour we should keep in mind if people have issues.